test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/IncludeSearchTests.cs

```diff
@@ -444,13 +478,702 @@ public async Task GivenARevIncludeSearchExpressionWithNoReferences_WhenSearched_
+        {
+            // Non-recursive iteration- Single iteration (_include:iterate)
+            string query = $"_include=MedicationDispense:prescription&_include:iterate=MedicationRequest:patient&_tag={Fixture.Tag}";
+
+            Bundle bundle = await Client.SearchAsync(ResourceType.MedicationDispense, query);
+
+            ValidateBundle(
+                bundle,
+                Fixture.AdamsMedicationDispense,
+                Fixture.SmithMedicationDispense,
+                Fixture.TrumanMedicationDispenseWithoutRequest,
+                Fixture.AdamsMedicationRequest,
+                Fixture.SmithMedicationRequest,
+                Fixture.AdamsPatient,
+                Fixture.SmithPatient);
+
+            ValidateSearchEntryMode(bundle, ResourceType.MedicationDispense);
+
+            ValidateBundleUrl(Client.HttpClient.BaseAddress, ResourceType.MedicationDispense, query, bundle.Link[0].Url);
+
+            // ensure that the included resources are not counted
+            bundle = await Client.SearchAsync(ResourceType.MedicationDispense, $"{query}&_summary=count");
+            Assert.Equal(3, bundle.Total);
+
+            // ensure that the included resources are not counted when _total is specified and the results fit in a single bundle.
+            bundle = await Client.SearchAsync(ResourceType.MedicationDispense, $"{query}&_total=accurate");
+            Assert.Equal(3, bundle.Total);
+        }
+
+        [Fact]
+        public async Task GivenAnIncludeRecurseSearchExpressionWithSingleIteration_WhenSearched_TheIterativeResultsShouldBeAddedToTheBundle()
+        {
+            // Non-recursive iteration- Single iteration (_include:recurse)
+            string query = $"_include=MedicationDispense:prescription&_include:recurse=MedicationRequest:patient&_tag={Fixture.Tag}";
+
+            Bundle bundle = await Client.SearchAsync(ResourceType.MedicationDispense, query);
+
+            ValidateBundle(
+                bundle,
+                Fixture.AdamsMedicationDispense,
+                Fixture.SmithMedicationDispense,
+                Fixture.TrumanMedicationDispenseWithoutRequest,
+                Fixture.AdamsMedicationRequest,
+                Fixture.SmithMedicationRequest,
+                Fixture.AdamsPatient,
+                Fixture.SmithPatient);
+
+            ValidateSearchEntryMode(bundle, ResourceType.MedicationDispense);
+
+            ValidateBundleUrl(Client.HttpClient.BaseAddress, ResourceType.MedicationDispense, query, bundle.Link[0].Url);
+        }
+
+        [Fact]
+        public async Task GivenAnIncludeIterateSearchExpressionWithAdditionalParameters_WhenSearched_TheIterativeResultsShouldBeAddedToTheBundle()
+        {
+            // Non-recursive iteration - Single iteration (_include:iterate)
+            // _tag parameter is not used due to issue https://github.com/microsoft/fhir-server/issues/1235
```

Comment:
This issue should be fixed now...

Reply:
BTW, the fix created another issue of getting invalid response for valid queries without _id parameter.
@eladiw  wrote to you about it

Reply:
Yeah, sorry about that.